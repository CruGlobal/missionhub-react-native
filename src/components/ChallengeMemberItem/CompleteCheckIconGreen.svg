<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M12 24C18.6274 24 24 18.6274 24 12C24 5.37258 18.6274 0 12 0C5.37258 0 0 5.37258 0 12C0 18.6274 5.37258 24 12 24Z" fill="#00CA99"/>
<path d="M17.3479 8.51172L11.5379 16.2577C11.4086 16.4294 11.2441 16.5714 11.0554 16.6741C10.8666 16.7769 10.6581 16.838 10.4437 16.8533C10.2294 16.8687 10.0143 16.8379 9.8128 16.7631C9.61135 16.6883 9.42827 16.5712 9.27588 16.4197L6.27588 13.4197" stroke="white" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
