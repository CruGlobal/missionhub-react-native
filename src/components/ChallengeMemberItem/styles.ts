import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  row: {
    paddingVertical: 16,
    paddingHorizontal: 20,
    width: theme.fullWidth,
  },
  nameText: {
    color: theme.primaryColor,
    fontSize: 16,
  },
  date: {
    fontSize: 12,
    color: theme.lightGrey,
  },
  card: {
    shadowColor: 'rgba(0,0,0,0.1)',
    shadowOffset: { width: 4, height: 6 },
    shadowOpacity: 0.3,
    shadowRadius: 2,
    elevation: 6,
    borderRadius: 8,
    marginVertical: 4,
  },
});
