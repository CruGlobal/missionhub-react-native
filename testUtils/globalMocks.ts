import faker from 'faker/locale/en';
import { IMocks } from 'graphql-tools';
import moment from 'moment';

import {
  CommunityCelebrationCelebrateableEnum,
  PermissionEnum,
  ReminderTypeEnum,
  StepTypeEnum,
} from '../__generated__/globalTypes';

let currentId = 1;
const nextId = () => currentId++;
export const resetGlobalMockSeeds = () => {
  currentId = 1;
  faker.seed(1);
};
resetGlobalMockSeeds();

export const globalMocks: IMocks = {
  String: () => faker.lorem.words(),
  Int: () => faker.random.number(),
  Float: () => faker.random.number({ precision: 0.01 }),
  Boolean: () => faker.random.boolean(),
  ID: () => nextId(),
  ISO8601DateTime: () => faker.date.past(10, '2020-01-14').toISOString(),
  ISO8601Date: () =>
    moment(faker.date.past(10, '2020-01-14')).format('YYYY-MM-DD'),

  BasePageInfo: () => ({
    endCursor: null,
    hasNextPage: false,
    hasPreviousPage: false,
    startCursor: null,
  }),
  ReminderTypeEnum: () => ReminderTypeEnum.once,
  StepTypeEnum: () => faker.random.arrayElement(Object.values(StepTypeEnum)),

  Step: () => ({
    title: faker.lorem.sentence(),
  }),
  Person: () => {
    const firstName = faker.name.firstName();
    const lastName = faker.name.lastName();
    return {
      fullName: `${firstName} ${lastName}`,
      firstName,
      lastName,
    };
  },
  Community: () => ({
    name: faker.company.catchPhrase(),
  }),
  CommunityCelebrationItem: () => {
    const firstName = faker.name.firstName();
    const lastName = faker.name.lastName();
    return {
      celebrateableType: faker.random.arrayElement(
        Object.values(CommunityCelebrationCelebrateableEnum),
      ),
      changedAttributeValue: moment(
        faker.date.past(10, '2020-01-14'),
      ).toISOString(),
      subjectPersonName: `${firstName} ${lastName}`,
    };
  },
  CommunityPermission: () => {
    return {
      permission: faker.random.arrayElement(Object.values(PermissionEnum)),
    };
  },
  FeedItem: () => {
    const firstName = faker.name.firstName();
    const lastName = faker.name.lastName();
    return {
      subjectPersonName: `${firstName} ${lastName}`,
    };
  },
};
