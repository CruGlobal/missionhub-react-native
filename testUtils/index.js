import 'react-native';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Need to import for all translation screens to work properly
import '../src/i18n';

Enzyme.configure({ adapter: new Adapter() });

export const createMockStore = (state = {}) => {
  return {
    getState: jest.fn(() => (state)),
    dispatch: jest.fn((response) => Promise.resolve(response)),
    subscribe: jest.fn(),
  };
};

export const createMockNavState = (params = {}) => {
  return { state: { params } };
};

export const testSnapshot = (data) => {
  expect(renderer.create(data)).toMatchSnapshot();
};

export const renderShallow = (component, store) => {
  let renderedComponent = shallow(
    component,
    { context: { store: store } }
  );

  // If component has translation wrappers, dive deeper
  while (renderedComponent.is('Translate') || renderedComponent.is('I18n')) {
    renderedComponent = renderedComponent.dive();
  }

  // Render contents of component
  renderedComponent = renderedComponent.dive();
  return renderedComponent;
};

export const testSnapshotShallow = (component, store) => {
  const renderedComponent = renderShallow(component, store);
  expect(renderedComponent).toMatchSnapshot();
  return renderedComponent;
};

export const mockFnWithParams = (obj, method, expectedReturn, ...expectedParams) => {
  return obj[method] = jest.fn(
    (...actualParams) => {
      expect(actualParams).toEqual(expectedParams);
      return expectedReturn;
    }
  );
};
