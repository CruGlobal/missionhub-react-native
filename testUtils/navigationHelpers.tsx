import React from 'react';
import {
  // eslint-disable-next-line import/named
  createAppContainer,
  // eslint-disable-next-line import/named
  createSwitchNavigator,
  // eslint-disable-next-line import/named
  NavigationParams,
  // eslint-disable-next-line import/named
  NavigationScreenProp,
} from 'react-navigation';
import { render } from 'react-native-testing-library';

export const createNavigationProp = (params?: NavigationParams) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let navigationProp: NavigationScreenProp<any> | undefined;

  const Navigator = createAppContainer(
    createSwitchNavigator({
      TestScreen: {
        screen: ({
          navigation,
        }: {
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          navigation: NavigationScreenProp<any>;
        }) => {
          navigationProp = navigation;
          return null;
        },
        params,
      },
    }),
  );
  render(<Navigator />);

  if (!navigationProp) {
    throw 'Unable to get navigation screen prop';
  }

  return navigationProp;
};
