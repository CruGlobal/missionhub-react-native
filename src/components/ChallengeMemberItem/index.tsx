import React from 'react';
import { Text } from 'react-native';
import moment from 'moment';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';

import { Flex, Touchable, Card, DateComponent } from '../common';
import { AuthState } from '../../reducers/auth';
import Avatar from '../Avatar';
import { ChallengeItem } from '../ChallengeStats';

import CompleteCheckIcon from './CompleteCheckIconGreen.svg';
import styles from './styles';

type PersonItem = {
  id: string;
  first_name: string;
  last_name?: string;
};

interface ChallengeMemberItemProps {
  item: ChallengeItem;
  date?: string;
  onSelect?: (person: PersonItem) => void;
}

const ChallengeMemberItem = ({
  date,
  item,
  onSelect,
}: ChallengeMemberItemProps) => {
  const { t } = useTranslation('challengeFeeds');
  const { completed_at, person } = item;
  const auth = useSelector(({ auth }: { auth: AuthState }) => auth);
  const isMe = person.id === auth.person.id;
  const todaysDate = moment().endOf('day');
  const isDateWithinWeek = todaysDate.isSame(moment(date).endOf('day'), 'week');

  const handleSelect = () => {
    onSelect && onSelect(person);
  };

  const content = (
    <Card style={styles.card}>
      <Flex align="center" direction="row" style={styles.row}>
        <Avatar
          style={{ marginRight: 10 }}
          // Use full_name for fullName until we convert these to use graphQL data
          person={{ ...person, fullName: person.full_name }}
          size={'medium'}
        />
        <Flex value={1} justify="between" direction="row" align="center">
          <Flex direction="column">
            <Text style={styles.nameText}>
              {isMe ? t('me') : person.full_name}
            </Text>
            <Text style={[styles.date]}>
              {completed_at ? `${t('completed')} ` : `${t('joined')} `}
              <DateComponent
                style={styles.date}
                date={date ? date : ''}
                format={isDateWithinWeek ? 'dddd' : 'dddd, LL'}
              />
            </Text>
          </Flex>
          <Flex style={{ marginRight: 25 }}>
            {completed_at ? <CompleteCheckIcon /> : null}
          </Flex>
        </Flex>
      </Flex>
    </Card>
  );

  if (onSelect) {
    return (
      <Touchable
        testID="ChallengeMemberItemButton"
        onPress={handleSelect}
        highlight={true}
      >
        {content}
      </Touchable>
    );
  }

  return content;
};

export default ChallengeMemberItem;
