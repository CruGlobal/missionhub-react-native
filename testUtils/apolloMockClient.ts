import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { addMockFunctionsToSchema, IMocks } from 'graphql-tools';
import { DocumentNode } from 'graphql';
import { buildClientSchema, IntrospectionQuery } from 'graphql/utilities';
import gql from 'fraql';
import { createMockerFromIntrospection } from 'fraql/mock';

import introspectionQuery from '../schema.json';

import { globalMocks } from './globalMocks';

export const createApolloMockClient = (mocks: IMocks = {}) => {
  const schema = buildClientSchema(
    (introspectionQuery as unknown) as IntrospectionQuery,
  );

  addMockFunctionsToSchema({
    schema,
    mocks: { ...globalMocks, ...mocks },
  });

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new SchemaLink({ schema }),
  });
};

export const mockFragment = <TData>(
  fragmentDocument: DocumentNode,
  options?: {
    mocks: IMocks;
  },
): TData => {
  if (!fragmentDocument.loc) {
    throw 'Unable to get original gql query to create FraQL fragment.';
  }

  return createMockerFromIntrospection(introspectionQuery, {
    mocks: globalMocks,
  }).mockFragment(gql(fragmentDocument.loc.source.body), options);
};
