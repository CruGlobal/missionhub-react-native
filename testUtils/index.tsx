import React, { ReactElement, ReactNode } from 'react';
import 'react-native';
import { Provider } from 'react-redux';
import { Provider as ProviderLegacy } from 'react-redux-legacy';
import thunk from 'redux-thunk';
import configureStore, { MockStore } from 'redux-mock-store';
// eslint-disable-next-line import/named
import { NavigationParams, NavigationProvider } from 'react-navigation';
import { ApolloProvider } from '@apollo/react-hooks';
import { ReactTestRendererJSON } from 'react-test-renderer';
import { render, RenderAPI } from 'react-native-testing-library';
import snapshotDiff from 'snapshot-diff';
import Enzyme, { shallow as enzymeShallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { IMocks } from 'graphql-tools';

import { createApolloMockClient } from './apolloMockClient';
import { createNavigationProp } from './navigationHelpers';

Enzyme.configure({ adapter: new Adapter() });

export const createThunkStore = configureStore([thunk]);

interface RenderWithContextParams {
  initialState?: {};
  store?: MockStore;
  navParams?: NavigationParams;
  mocks?: IMocks;
  noWrappers?: boolean;
}

// Inspiration from https://github.com/kentcdodds/react-testing-library/blob/52575005579307bcfbe7fbe4ef4636147c03c6fb/examples/__tests__/react-redux.js#L69-L80
export function renderWithContext(
  component: ReactElement,
  {
    initialState,
    store = createThunkStore(initialState),
    navParams,
    mocks: mocks = {},
    noWrappers = false,
  }: RenderWithContextParams = {},
) {
  const mockApolloClient = createApolloMockClient(mocks);

  const navigation = createNavigationProp(navParams);

  let renderResult: RenderAPI;
  if (noWrappers) {
    renderResult = render(React.cloneElement(component, { navigation }));
  } else {
    // Warning: don't call any functions in here that return new instances on every call. All the props need to stay the same otherwise rerender won't work.
    const wrapper = ({ children }: { children: ReactNode }) => (
      <NavigationProvider value={navigation}>
        <ProviderLegacy store={store}>
          <Provider store={store}>
            <ApolloProvider client={mockApolloClient}>
              {children}
            </ApolloProvider>
          </Provider>
        </ProviderLegacy>
      </NavigationProvider>
    );

    renderResult = render(React.cloneElement(component, { navigation }), {
      wrapper,
    });
  }

  let storedSnapshot: ReactTestRendererJSON | null;
  return {
    ...renderResult,
    store,
    snapshot: () => expect(renderResult.toJSON()).toMatchSnapshot(),
    recordSnapshot: () => (storedSnapshot = renderResult.toJSON()),
    diffSnapshot: () => {
      if (!storedSnapshot) {
        throw new Error(
          'You must call recordSnapshot to store an initial snapshot before calling diffSnapshot',
        );
      }
      expect(
        snapshotDiff(storedSnapshot, renderResult.toJSON()),
      ).toMatchSnapshot();
    },
  };
}

// TODO: Remove all legacy rendering functions below

export const createMockNavState = (params = {}) => {
  return { state: { params } };
};

export const testSnapshot = (component: ReactElement) => {
  const { toJSON } = render(component);
  expect(toJSON()).toMatchSnapshot();
};

export const renderShallow = (
  component: ReactElement,
  store = createThunkStore(),
) => {
  let renderedComponent = enzymeShallow(
    <ProviderLegacy store={store}>{component}</ProviderLegacy>,
  ).dive();

  // If component has translation wrappers, dive deeper
  while (
    // eslint-disable-next-line  @typescript-eslint/no-explicit-any
    ((renderedComponent.type() as any).displayName || '').startsWith(
      'withI18nextTranslation(',
    )
  ) {
    renderedComponent = renderedComponent.dive();
  }

  // Render contents of component
  renderedComponent = renderedComponent.dive();
  return renderedComponent;
};

export const testSnapshotShallow = (
  component: ReactElement,
  store = createThunkStore(),
) => {
  const renderedComponent = renderShallow(component, store);
  expect(renderedComponent).toMatchSnapshot();
  return renderedComponent;
};
