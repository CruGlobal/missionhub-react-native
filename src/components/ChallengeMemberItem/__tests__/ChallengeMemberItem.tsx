import React from 'react';
import MockDate from 'mockdate';
import { fireEvent } from 'react-native-testing-library';

import { renderWithContext } from '../../../../testUtils';
import ChallengeMemberItem from '..';

const mockDate = '2020-02-29 12:00:00 PM GMT+0';
MockDate.set(mockDate);

const date = '2020-02-26';
const myId = '1234';
const item = {
  id: '1',
  title: 'Cool Challenge',
  accepted_at: date,
  end_date: '2020-03-01',
  accepted_count: 1,
  completed_count: 0,
  person: {
    id: myId,
    full_name: 'Christian Huffman',
  },
};
const onSelect = jest.fn();
const props = {
  date,
  onSelect,
  item,
};

const completedItem = {
  ...item,
  completed_at: '2020-02-27',
};

it('render correctly | Me', () => {
  renderWithContext(<ChallengeMemberItem {...props} />, {
    initialState: {
      auth: {
        person: {
          id: myId,
        },
      },
    },
  }).snapshot();
});

it('renders correctly | Not Me', () => {
  renderWithContext(<ChallengeMemberItem {...props} />, {
    initialState: {
      auth: {
        person: {
          id: '4',
        },
      },
    },
  }).snapshot();
});

it('renders completed item correctly', () => {
  renderWithContext(
    <ChallengeMemberItem
      item={completedItem}
      date={date}
      onSelect={onSelect}
    />,
    {
      initialState: {
        auth: {
          person: {
            id: myId,
          },
        },
      },
    },
  ).snapshot();
});

it('renders full date if not in same week', () => {
  const dateItem = {
    ...item,
    accepted_at: '2020-02-22',
  };
  renderWithContext(
    <ChallengeMemberItem
      item={dateItem}
      date={'2020-02-22'}
      onSelect={onSelect}
    />,
    {
      initialState: {
        auth: {
          person: {
            id: myId,
          },
        },
      },
    },
  ).snapshot();
});

it('fires onSelect when item is pressed', async () => {
  const { getByTestId } = renderWithContext(
    <ChallengeMemberItem {...props} />,
    {
      initialState: {
        auth: {
          person: {
            id: myId,
          },
        },
      },
    },
  );

  await fireEvent.press(getByTestId('ChallengeMemberItemButton'));
  expect(onSelect).toHaveBeenCalledWith(item.person);
});
