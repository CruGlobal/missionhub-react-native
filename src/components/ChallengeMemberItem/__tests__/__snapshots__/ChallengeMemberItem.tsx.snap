// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly | Me 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID="ChallengeMemberItemButton"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "elevation": 4,
          "flexDirection": "row",
          "marginHorizontal": 16,
          "marginVertical": 8,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        },
        Object {
          "borderRadius": 8,
          "elevation": 6,
          "marginVertical": 4,
          "shadowColor": "rgba(0,0,0,0.1)",
          "shadowOffset": Object {
            "height": 6,
            "width": 4,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 16,
            "width": 750,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Array [
              Object {
                "borderRadius": 24,
                "height": 48,
                "width": 48,
              },
              Object {
                "backgroundColor": "rgb(192, 136, 125)",
              },
              Object {
                "marginRight": 10,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "fontSize": 26,
                  "fontWeight": "300",
                },
              ],
            ]
          }
        >
          C
        </Text>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Object {
                  "color": "#007398",
                  "fontSize": 16,
                },
              ]
            }
          >
            Me
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Array [
                  Object {
                    "color": "#B4B6BA",
                    "fontSize": 12,
                  },
                ],
              ]
            }
          >
            Joined 
            <Text
              style={
                Array [
                  Object {
                    "color": "#505256",
                  },
                  Object {
                    "fontFamily": "SourceSansPro-Regular",
                  },
                  Object {
                    "color": "#B4B6BA",
                    "fontSize": 12,
                  },
                ]
              }
              testID="Text"
            >
              Wednesday
            </Text>
          </Text>
        </View>
        <Image
          source={null}
          style={
            Object {
              "marginRight": 25,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders completed item correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID="ChallengeMemberItemButton"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "elevation": 4,
          "flexDirection": "row",
          "marginHorizontal": 16,
          "marginVertical": 8,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        },
        Object {
          "borderRadius": 8,
          "elevation": 6,
          "marginVertical": 4,
          "shadowColor": "rgba(0,0,0,0.1)",
          "shadowOffset": Object {
            "height": 6,
            "width": 4,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 16,
            "width": 750,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Array [
              Object {
                "borderRadius": 24,
                "height": 48,
                "width": 48,
              },
              Object {
                "backgroundColor": "rgb(192, 136, 125)",
              },
              Object {
                "marginRight": 10,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "fontSize": 26,
                  "fontWeight": "300",
                },
              ],
            ]
          }
        >
          C
        </Text>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Object {
                  "color": "#007398",
                  "fontSize": 16,
                },
              ]
            }
          >
            Me
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Array [
                  Object {
                    "color": "#B4B6BA",
                    "fontSize": 12,
                  },
                ],
              ]
            }
          >
            Completed 
            <Text
              style={
                Array [
                  Object {
                    "color": "#505256",
                  },
                  Object {
                    "fontFamily": "SourceSansPro-Regular",
                  },
                  Object {
                    "color": "#B4B6BA",
                    "fontSize": 12,
                  },
                ]
              }
              testID="Text"
            >
              Wednesday
            </Text>
          </Text>
        </View>
        <Image
          source={
            Object {
              "testUri": "../../../assets/images/challengeCompleteGreen.png",
            }
          }
          style={
            Object {
              "marginRight": 25,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly | Not Me 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID="ChallengeMemberItemButton"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "elevation": 4,
          "flexDirection": "row",
          "marginHorizontal": 16,
          "marginVertical": 8,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        },
        Object {
          "borderRadius": 8,
          "elevation": 6,
          "marginVertical": 4,
          "shadowColor": "rgba(0,0,0,0.1)",
          "shadowOffset": Object {
            "height": 6,
            "width": 4,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 16,
            "width": 750,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Array [
              Object {
                "borderRadius": 24,
                "height": 48,
                "width": 48,
              },
              Object {
                "backgroundColor": "rgb(192, 136, 125)",
              },
              Object {
                "marginRight": 10,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "fontSize": 26,
                  "fontWeight": "300",
                },
              ],
            ]
          }
        >
          C
        </Text>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Object {
                  "color": "#007398",
                  "fontSize": 16,
                },
              ]
            }
          >
            Christian Huffman
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Array [
                  Object {
                    "color": "#B4B6BA",
                    "fontSize": 12,
                  },
                ],
              ]
            }
          >
            Joined 
            <Text
              style={
                Array [
                  Object {
                    "color": "#505256",
                  },
                  Object {
                    "fontFamily": "SourceSansPro-Regular",
                  },
                  Object {
                    "color": "#B4B6BA",
                    "fontSize": 12,
                  },
                ]
              }
              testID="Text"
            >
              Wednesday
            </Text>
          </Text>
        </View>
        <Image
          source={null}
          style={
            Object {
              "marginRight": 25,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders full date if not in same week 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
  testID="ChallengeMemberItemButton"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderRadius": 8,
          "elevation": 4,
          "flexDirection": "row",
          "marginHorizontal": 16,
          "marginVertical": 8,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        },
        Object {
          "borderRadius": 8,
          "elevation": 6,
          "marginVertical": 4,
          "shadowColor": "rgba(0,0,0,0.1)",
          "shadowOffset": Object {
            "height": 6,
            "width": 4,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 2,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 16,
            "width": 750,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Array [
              Object {
                "borderRadius": 24,
                "height": 48,
                "width": 48,
              },
              Object {
                "backgroundColor": "rgb(192, 136, 125)",
              },
              Object {
                "marginRight": 10,
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#505256",
              },
              Object {
                "fontFamily": "SourceSansPro-Regular",
              },
              Array [
                Object {
                  "color": "#ffffff",
                },
                Object {
                  "fontSize": 26,
                  "fontWeight": "300",
                },
              ],
            ]
          }
        >
          C
        </Text>
      </View>
      <View
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {},
              Object {
                "flexDirection": "column",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Object {
                  "color": "#007398",
                  "fontSize": 16,
                },
              ]
            }
          >
            Me
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#505256",
                },
                Object {
                  "fontFamily": "SourceSansPro-Regular",
                },
                Array [
                  Object {
                    "color": "#B4B6BA",
                    "fontSize": 12,
                  },
                ],
              ]
            }
          >
            Joined 
            <Text
              style={
                Array [
                  Object {
                    "color": "#505256",
                  },
                  Object {
                    "fontFamily": "SourceSansPro-Regular",
                  },
                  Object {
                    "color": "#B4B6BA",
                    "fontSize": 12,
                  },
                ]
              }
              testID="Text"
            >
              Saturday, February 22, 2020
            </Text>
          </Text>
        </View>
        <Image
          source={null}
          style={
            Object {
              "marginRight": 25,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
